ArrayList/ArrayList.cs

```diff
@@ -1,8 +1,580 @@
 ﻿using System;
+using System.Text;
 
-namespace ArrayList
+namespace List
 {
-    public class ArrayList
+    public class ArrayList<T> where T : IComparable<T>
     {
+        public int Length
+        {
+            get
+            {
+                return _length;
+            }
+            private set
+            {
+                if (value >= 0)
+                {
+                    _length = value;
+                }
+                else if (value < 0)
+                {
+                    _length = 0;
+                }
+            }
+        }
+
+        public T this[int index]
+        {
+            get
+            {
+                if (index >= 0 && index < Length)
+                {
+                    return _array[index];
+                }
+
+                throw new IndexOutOfRangeException();
+            }
+            set
+            {
+                if (index >= 0 && index < _array.Length)
+                {
+                    _array[index] = value;
+                }
+
+                throw new IndexOutOfRangeException();
+            }
+        }
+
+        private int _length;
```

Comment:
Fields should be in the beginning of class.

Reply:
ok